File: src/main/java/cn/dsnbo/bedrockplayersupport/listener/login/NexAuthListener.java
package cn.dsnbo.bedrockplayersupport.listener.login;

import cn.dsnbo.bedrockplayersupport.BedrockPlayerSupport;
import cn.dsnbo.bedrockplayersupport.config.Config;
import cn.dsnbo.bedrockplayersupport.config.Language;
import cn.dsnbo.bedrockplayersupport.util.StringUtil;
import org.bukkit.ChatColor;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;
import org.geysermc.floodgate.api.FloodgateApi;
import su.nexmedia.auth.NexAuthAPI;
import su.nexmedia.auth.auth.impl.AuthPlayer;
import su.nexmedia.auth.auth.impl.PlayerState;

/**
 * @author DongShaoNB
 */
public class NexAuthListener implements Listener {

  @EventHandler
  public void onPlayerJoinEvent(PlayerJoinEvent event) {
    Player player = event.getPlayer();
    Config config = BedrockPlayerSupport.getMainConfigManager().getConfigData();
    Language language = BedrockPlayerSupport.getLanguageConfigManager().getConfigData();
    if (FloodgateApi.getInstance().isFloodgatePlayer(player.getUniqueId())) {
      if (AuthPlayer.getOrCreate(player).isRegistered()) {
        if (AuthPlayer.getOrCreate(player).getState() == PlayerState.IN_LOGIN) {
          if (config.enableLogin()) {
            NexAuthAPI.getAuthManager().login(player);
            player.sendMessage(
                BedrockPlayerSupport.getMiniMessage().deserialize(language.loginSuccessfully()));
          }
        }
      } else {
        if (config.enableRegister()) {
          String password = StringUtil.randomString(config.passwordLength());
          NexAuthAPI.getAuthManager().register(player, password);
          player.sendMessage(
                BedrockPlayerSupport.getMiniMessage().deserialize(language.registerSuccessfully()
                    .replaceAll("%password%", password)));
        }
      }
    }
  }
}

File: src/main/kotlin/cc/dsnb/bedrockplayersupport/listener/auth/NexAuthListener.kt
package cc.dsnb.bedrockplayersupport.listener.auth

import cc.dsnb.bedrockplayersupport.BedrockPlayerSupport
import cc.dsnb.bedrockplayersupport.util.StringUtil
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent
import su.nexmedia.auth.NexAuthAPI
import su.nexmedia.auth.auth.impl.AuthPlayer
import su.nexmedia.auth.auth.impl.PlayerState

/**
 * @author DongShaoNB
 */
class NexAuthListener : Listener {

    @EventHandler
    fun onPlayerJoin(event: PlayerJoinEvent) {
        val player = event.player
        val mainConfig = BedrockPlayerSupport.mainConfigManager.getConfigData()
        val langConfig = BedrockPlayerSupport.langConfigManager.getConfigData()
        if (BedrockPlayerSupport.floodgateApi.isFloodgatePlayer(player.uniqueId)) {
            if (AuthPlayer.getOrCreate(player).isRegistered) {
                if (AuthPlayer.getOrCreate(player).state == PlayerState.IN_LOGIN) {
                    if (mainConfig.enableLogin()) {
                        NexAuthAPI.getAuthManager().login(player)
                        player.sendMessage(
                            BedrockPlayerSupport.miniMessage.deserialize(langConfig.loginSuccessfully())
                        )
                    }
                }
            } else {
                if (mainConfig.enableRegister()) {
                    val password = StringUtil.randomString(mainConfig.passwordLength())
                    NexAuthAPI.getAuthManager().register(player, password)
                    player.sendMessage(
                        BedrockPlayerSupport.miniMessage.deserialize(
                            langConfig.registerSuccessfully()
                                .replace("%password%", password)
                        )
                    )
                }
            }
        }
    }

}